File: client/src/services/categoryService.js
import api from './api';

const categoryService = {
    getAllCategories: async () => {
        const response = await api.get('/api/categories');
        return response.data;
    },

    getCategoryBySlug: async (slug) => {
        const response = await api.get(`/api/categories/${slug}`);
        return response.data;
    }
};

export default categoryService;

File: server/src/main/java/com/fptmarket/service/CategoryService.java
package com.fptmarket.service;

import com.fptmarket.dto.request.CategoryRequest;
import com.fptmarket.dto.response.CategoryResponse;

import java.util.List;

public interface CategoryService {
    List<CategoryResponse> getAllCategories();
    CategoryResponse getCategoryBySlug(String slug);
    CategoryResponse createCategory(CategoryRequest request);
    CategoryResponse updateCategory(Long id, CategoryRequest request);
    void deleteCategory(Long id);
}
